options.go

```diff
@@ -20,19 +37,42 @@ var defaults = func(c *config) error {
 	c.refreshInterval = 15 * time.Minute
 	c.requestTimeout = 30 * time.Second
 
+	c.dialTimeout = 15 * time.Second
+	c.maxPeerAddresses = 16
+	c.throttleGlobalMax = 30
+	c.throttlePeerMax = 3
+	c.throttleResetPeriod = 1 * time.Minute
+	c.throttleResetJitter = 15 * time.Second
 	return nil
 }
 
-// WithAddresses allows overriding which Addresses the AutoNAT client beliieves
+// EnableService specifies that AutoNAT should be allowed to run a NAT service to help
+// other peers determine their own NAT status. The provided Network should not be the
+// default network/dialer of the host passed to `New`, as the NAT system will need to
+// make parallel connections, and as such will modify both the associated peerstore
+// and terminate connections of this dialer. The dialer provided
+// should be compatible (TCP/UDP) however with the transports of the libp2p network.
+func EnableService(dialer network.Network, forceServer bool) Option {
+	return func(c *config) error {
+		if dialer == c.host.Network() || dialer.Peerstore() == c.host.Peerstore() {
+			return errors.New("dialer should not be that of the host")
+		}
+		c.dialer = dialer
+		c.forceServer = forceServer
+		return nil
+	}
+}
+
+// AddressGuesser allows overriding which Addresses the AutoNAT client believes
 // are "its own". Useful for testing, or for more exotic port-forwarding
 // scenarios where the host may be listening on different ports than it wants
 // to externally advertise or verify connectability on.
-func WithAddresses(addrFunc GetAddrs) Option {
+func AddressGuesser(addrFunc AddrFunc) Option {
```

Comment:
Hm. I now hate myself, that was a bad suggestion. Can you think of a better alternative?